### PR TITLE
stream blocks

### PR DESCRIPTION
**Pull request by anjor**
For https://github.com/rpcpool/yellowstone-faithful/issues/173

### CODE REVIEW

File: old-faithful-proto/proto/old-faithful.proto [hunk 1]
```diff
@@ -90,3 +92,34 @@ message GetResponseError {
   GetResponseErrorCode code = 1;
   string message = 2;
 }
+
+
+message StreamBlocksRequest {
+  uint64 start_slot = 1;
+  optional uint64 end_slot = 2;  // If not specified, continues indefinitely
```
Comment (gagliardetto) [inline old-faithful-proto/proto/old-faithful.proto:99]: Maybe let's put a max value? 100? 1k?

File: old-faithful-proto/proto/old-faithful.proto [hunk 2]
```diff
@@ -90,3 +92,34 @@ message GetResponseError {
...
+  optional StreamBlocksFilter filter = 3;
+}
+
+message StreamBlocksFilter {
+  repeated string account_include = 1;  // Filter blocks/txns mentioning these accounts
+  optional bool include_transactions = 2;  // Include full transaction details
+  optional bool include_accounts = 3;  // Include account updates
```
Comment (gagliardetto) [inline old-faithful-proto/proto/old-faithful.proto:106]: what does this do?

Comment (anjor) [reply old-faithful-proto/proto/old-faithful.proto:106]: I was just following the dragon's mouth api: https://docs.triton.one/project-yellowstone/dragons-mouth-grpc-subscriptions#subscribe-to-blocks

Comment (gagliardetto) [reply old-faithful-proto/proto/old-faithful.proto:106]: do you know if that means "filter transactions that include these accounts" or "actually include the actual accounts state"

File: grpc-server.go [hunk 3]
```diff
@@ -621,3 +624,90 @@ func (multi *MultiEpoch) GetBlockTime(ctx context.Context, params *old_faithful_
...
+			continue
+		}
+
+		for _, acc := range solTx.Message.AccountKeys {
+			if _, exists := accountSet[acc.String()]; exists {
+				return true
+			}
+		}
+
+		for _, inst := range solTx.Message.Instructions {
```
Comment (gagliardetto) [inline grpc-server.go:695]: this seems to be redundant

Comment (anjor) [reply grpc-server.go:695]: So no loop required over instructions?

Comment (gagliardetto) [reply grpc-server.go:695]: all the accounts are in the tx. message 

unless you consider also the accounts from address lookup tables, and in that case you need to get that from tx meta

Comment (linuskendall) [reply grpc-server.go:695]: Let’s not support address table lookups for now but add it as a future potential feature  (add a ticket for it ).

Comment (anjor) [reply grpc-server.go]: done!

**Pull request merged**